compiler/rustc_parse/src/parser/path.rs

```diff
@@ -934,6 +941,9 @@ impl<'a> Parser<'a> {
             }
         } else if self.token.is_keyword(kw::Const) {
             return self.recover_const_param_declaration(ty_generics);
+        } else if let Some(attr_span) = attr_span {
+            let diag = self.dcx().create_err(AttributeOnEmptyType { span: attr_span });
+            return Err(diag);
         } else {
             // Fall back by trying to parse a const-expr expression. If we successfully do so,
             // then we should report an error that it needs to be wrapped in braces.
```

Comment:
Two lines below, there's a preexisting `let attrs = self.parse_outer_attributes()?;`. We should be able to drop it and pass the already parsed attrs to the `parse_expr_res`. 

It's not blocking concern tho ig and it would stop us from recovering from `///` and `/** */` on gen args. Can be done in a follow-up.